sds/src/scanner/mod.rs

```diff
@@ -163,7 +163,11 @@ impl Scanner {
         ScannerBuilder::new(rules)
     }
 
-    pub fn scan<E: Event>(&self, event: &mut E) -> Vec<RuleMatch> {
+    pub fn scan<E: Event>(
+        &self,
+        event: &mut E,
+        blocked_rules_idx: Option<Vec<usize>>,
```

Comment:
It does not seem to have difference of behaviour between `None` and `Some(vec![])`. In that case, the `None` state does not add more information than an empty list. To simplify our interface, I suggest to drop the `Option` and keep only the `Vec<usize>`. It will reduce the indirection of the parameter.

Reply:
faire enough

Reply:
A `Vec` doesn't allocate memory until something it pushed into it, so it should be fine to use an empty `Vec` instead of `None`. (Although this is usually passed in as a slice instead of `Vec` to be more generic, so `blocked_rules_idx: &[usize]`)